compiler/quilt/tools/command.py

```diff
@@ -1129,3 +1129,18 @@ def _print_node(node, prefix, child_prefix, name, path):
 
     print(pkgobj.get_path())
     _print_children(children=pkgobj.get_contents().children.items(), prefix='', path='')
+
+def rm(package, force=False):
+    """
+    Remove a package (all instances) from the local store.
+    """
+    owner, pkg = parse_package(package)
+
+    if not force:
+        confirmed = input("Remove {0}? (y/n)".format(package))
+        if confirmed.lower() != 'y':
+            return
+
+    store = PackageStore()
+    store.remove_package(owner, pkg)
+    store.prune()
```

Comment:
I'm not convinced that automatic pruning is the right thing to do:
1) Logically, they're two separate actions. `rm` is per package; pruning is not. You may want to prune even without deleting a single package - e.g., after rebuilding a package a bunch of times.
2) This could have unexpected results: e.g., delete objects from a large download that got interrupted.

(One workaround is to collect all used objects, then delete the package, then collect them again, and only delete the newly-orphaned objects... Though not sure it's worth the effort.)

Reply:
I think you're right that we should avoid surprises and not simply call prune as part of rm. I've added an input set to prune that will limit the pruning to only that set of objects. Now, remove collects all the objects referenced in the package that's being deleted and passes that set to prune. That way, only objects from the removed package are eligible for deletion.